waffle/onnx/onnxgraph.py

```diff
@@ -1,13 +1,70 @@
 from typing import List
-from waffle.onnx.node import Node
-
+from waffle import tensor
+from waffle.onnx.node import Node  
+import os
 
+WFLDBG = os.environ.get('WFLDBG')
+#edit spaces for 2
 class onnxGraph:
-  def __init__(self, list:List[Node]):
-    self.liniearized_graph = list
+    def __init__(self, linearized_list):
+        self.nodes = [Node(node.name, node.input, node.output, node.op_type, node.attributes, node.params) for node in linearized_list]
+        self.build_graph()
+        self.hard_traverse()
+
+    def build_graph(self):
+        for i, node in enumerate(self.nodes):
+            node.input_indices = []
+            for input_name in node.input:
+                for j, other_node in enumerate(self.nodes):
+                    if input_name in other_node.output:
+                        node.input_indices.append(j)    
+
+    def hard_traverse(self):
+        for i, node in enumerate(self.nodes):
+            inputs = []
+            if i == 0:
+                inputs.append(-1)
+            for input_index in node.input_indices:
+                inputs.append(input_index)
+
+            node.set_traverse_input(inputs)
+            node.search_layer()
+            
+    def run(self, input):
```

Comment:
dfs/bfs/Dijkstra should be here to traverse the graph! Before that finish constructing the graph!